### PR TITLE
Remove ingress resources

### PR DESCRIPTION
**Pull request by sslavic**
This PR is part of removing IC as built-in component, making it optional app.

Related: https://github.com/giantswarm/giantswarm/issues/7029

### CODE REVIEW

**Comment by rossf7** [comment]
@sslavic This LGTM and matches my understanding of what needs to be removed.

Not approving yet until you've had a chance to test the deletion process.

**Comment by corest** [comment]
Crap, I by mistake created the same branch and pushed it to your branch @sslavic :D

**Comment by sslavic** [comment]
> Crap, I by mistake created the same branch and pushed it to your branch @sslavic :D

No prob. If you agree I can force push my local branch, and you can push your code to a new branch (and track new branch).

**Comment by corest** [comment]
@sslavic I did a copy branch for my tests. You can overwrite this one

**Comment by sslavic** [comment]
Tested successfully on `ginger` with script and support provided by @tuommaki / @MarcelMue.
@rossf7 PTAL

**Comment by rossf7** [comment]
This looks good but the flag to disable the ingress controller service should be true.

https://github.com/giantswarm/aws-operator/blob/4d4c50948e96613f11f5ebaf366077ab755cc8a2/service/controller/clusterapi/v31/cloudconfig/tccp.go#L77-L79

This is where it's used in the ignition.

https://github.com/giantswarm/k8scloudconfig/blob/2ae58d542ab23c7abd27ffd2366878a1a82218cc/v_4_9_0/master_template.go#L393-L399

File: service/controller/clusterapi/v30/cloudconfig/tccp.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func (t *TCCP) Render(ctx context.Context, cr cmav1alpha1.Cluster, clusterCerts
 		params.DisableEncryptionAtREST = true
 		// Ingress controller service remains in k8scloudconfig and will be
 		// removed in a later migration.
-		params.DisableIngressControllerService = false
+		params.DisableIngressControllerService = true
```
Comment (rossf7) [inline service/controller/clusterapi/v30/cloudconfig/tccp.go]: We should also update the comment. Sorry I missed that first time.


> Ingress Controller service is not created via ignition. It is created by the Ingress Controller app if it is installed in the tenant cluster.

Comment (rossf7) [reply service/controller/clusterapi/v30/cloudconfig/tccp.go]: Damn looks like it also affected the golden file test since this manifest is removed.

**Comment by xh3b4sd** [comment]
Please be aware of https://github.com/giantswarm/aws-operator/pull/1963. You heavily work on kind of the same code now as it seems.

File: service/controller/clusterapi/v30/cloudconfig/tccp.go (hunk 1 above)
Comment (sslavic) [reply service/controller/clusterapi/v30/cloudconfig/tccp.go]: yep, fixing tests, thanks for comment text too!

**Comment by tomahawk28** [comment]
`Test_Controller_CloudConfig_TCCP_Template_Render` is failed for some reasons.

**Comment by sslavic** [comment]
This PR shouldn't be merged until https://github.com/giantswarm/aws-operator/pull/1963 is done.
For more details see https://gigantic.slack.com/archives/C2MS2VB37/p1571223384060400

**Comment by TheoBrigitte** [comment]
`clusterapi/v30` was completely dropped as part of flattening.

**Comment by sslavic** [comment]
Retested on ginger after flatting of aws-operator. @rossf7 PTAL

File: service/controller/clusterapi/v31/adapter/guest_load_balancers.go [hunk 2]
```diff
@@ -2,6 +2,7 @@ package adapter
 
 import (
 	"fmt"
+
```
Comment (rossf7) [inline service/controller/clusterapi/v31/adapter/guest_load_balancers.go]: <3

File: service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go [hunk 3]
```diff
@@ -67,25 +67,6 @@ const TemplateMainRecordSets = `
...
-    Type: AWS::Route53::RecordSet
-    Properties:
-      AliasTarget:
-        DNSName: !GetAtt IngressLoadBalancer.DNSName
-        HostedZoneId: !GetAtt IngressLoadBalancer.CanonicalHostedZoneNameID
-        EvaluateTargetHealth: false
-      Name: 'ingress.{{ $v.ClusterID }}.k8s.{{ $v.BaseDomain }}.'
-      HostedZoneId: !Ref 'HostedZone'
-      Type: A
-  IngressWildcardRecordSet:
```
Comment (corest) [inline service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go]: As discussed here https://gigantic.slack.com/archives/C3S482Z98/p1571996691134200, this wildcard record should be kept.

File: service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go [hunk 4]
```diff
@@ -67,25 +67,6 @@ const TemplateMainRecordSets = `
...
-    Type: AWS::Route53::RecordSet
-    Properties:
-      Name: '*.{{ $v.ClusterID }}.k8s.{{ $v.BaseDomain }}.'
-      HostedZoneId: !Ref 'HostedZone'
-      TTL: '300'
-      Type: CNAME
-      ResourceRecords:
-        - !Ref 'IngressRecordSet'
```
Comment (corest) [inline service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go]: Then this needs to be fixed with `'*.{{ $v.ClusterID }}.k8s.{{ $v.BaseDomain }}.'`
@sslavic fyi

File: service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go (hunk 3 above)
Comment (sslavic) [reply service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go]: Fixed, PTAL

File: service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go (hunk 4 above)
Comment (sslavic) [reply service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go]: Done, PTAL

File: service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go [hunk 5]
```diff
@@ -67,6 +67,15 @@ const TemplateMainRecordSets = `
...
       Type: A
+  IngressWildcardRecordSet:
+    Type: AWS::Route53::RecordSet
+    Properties:
+      Name: '*.{{ $v.ClusterID }}.k8s.{{ $v.BaseDomain }}.'
+      HostedZoneId: !Ref 'HostedZone'
+      TTL: '300'
+      Type: CNAME
+      ResourceRecords:
+        - '*.{{ $v.ClusterID }}.k8s.{{ $v.BaseDomain }}.'
```
Comment (corest) [inline service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go]: it is `ingress.<domain>`, not `wildcard`

Comment (sslavic) [reply service/controller/clusterapi/v31/resource/tccp/template/template_main_record_sets.go]: Should it be renamed to `IngressRecordSet`?

**Pull request merged**